src/main/scala/com/fulcrumgenomics/util/SimpleCounter.scala

```diff
@@ -39,17 +37,23 @@ object SimpleCounter {
   * zero for any item that has not been counted yet.  
   */
 class SimpleCounter[T] extends Iterable[(T, Long)] {
-  private val counts = mutable.Map[T,Long]()
+  import scala.collection.JavaConversions._
+
+  private val counts = new java.util.TreeMap[T,Long]()
+  private var _totalCounts: Long = 0L
 
   /** Increment the count of object T by 1. */
   def count(t: T): Unit = count(t, 1)
 
   /** Increment the count of object T by the specified value. */
-  def count(t: T, n: Long): Unit = counts(t) = counts.getOrElse(t, 0L) + n
+  def count(t: T, n: Long): Unit = { _totalCounts += n; counts(t) = counts.getOrDefault(t, 0L) + n }
```

Comment:
Is this going to cause multiple implicit conversions every time count is called?